src/main/java/frc/robot/subsystems/climber/Climber.java

```diff
@@ -0,0 +1,121 @@
+package frc.robot.subsystems.climber;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.subsystems.UnitModel;
+
+public class Climber extends SubsystemBase {
+    private final TalonFX motor = new TalonFX(Ports.Climber.MOTOR);
+    private final TalonFX salve = new TalonFX(Ports.Climber.SLAVE);
+    private final UnitModel unitModel = new UnitModel(Constants.Climber.TICKS_PER_METER);
+    private final Solenoid stopper = new Solenoid(Ports.Climber.STOPPER);
+    private final Solenoid gearbox = new Solenoid(Ports.Climber.GEARBOX);
+
+    public Climber() {
+        salve.follow(motor);
+
+        motor.setInverted(Ports.Climber.MOTOR_INVERTED);
+        salve.setInverted(Ports.Climber.SLAVE_INVERTED);
+
+        motor.setSensorPhase(Ports.Climber.MOTOR_SENSOR_PHASE_INVERTED);
+        salve.setSensorPhase(Ports.Climber.SLAVE_SENSOR_PHASE_INVERTED);
+
+        motor.config_kP(0, Constants.Climber.kP);
+        motor.config_kI(0, Constants.Climber.kI);
+        motor.config_kD(0, Constants.Climber.kD);
+
+        salve.config_kP(0, Constants.Climber.kP);
+        salve.config_kI(0, Constants.Climber.kI);
+        salve.config_kD(0, Constants.Climber.kD);
+    }
+
+    /**
+     * Get the climber's height.
+     *
+     * @return the climber's height [double].
+     */
+    public double getHeight() {
+        return unitModel.toUnits(motor.getSelectedSensorPosition());
+    }
+
+    /**
+     * Set the climber's height.
+     *
+     * @param height height [double].
+     */
+    public void setHeight(double height) {
+        motor.set(ControlMode.Position, unitModel.toTicks(height));
+    }
+
+    /**
+     * Get the stopper mode.
+     *
+     * @return the stopper mode [PistonMode].
+     */
+    public PistonMode getStopperMode() {
+        if (stopper.get())
+            return PistonMode.OPEN;
+        else
+            return PistonMode.CLOSED;
+    }
+
+    /**
+     * Get the gearbox mode.
+     *
+     * @return the gearbox mode [PistonMode].
+     */
+    public PistonMode getGearboxMode() {
+        if (gearbox.get())
+            return PistonMode.OPEN;
+        else
+            return PistonMode.CLOSED;
+    }
+
+    /**
+     * Set the stopper mode.
+     *
+     * @param mode the stopper mode [boolean].
+     */
+    public void setStopperMode(boolean mode) {
+        stopper.set(mode);
+    }
+
+    /**
+     * Set the gearbox mode.
+     *
+     * @param mode the gearbox mode [boolean].
+     */
+    public void setGearboxMode(boolean mode) {
+        gearbox.set(mode);
+    }
+
+    /**
+     * Enum for piston mode, OPEN is true, CLOSED is false.
+     */
+    public enum PistonMode {

```

Comment:
Put also a toggle mode

Reply:
Is toggling a mode of a piston? IMO It's an action, but not a mode.